include/qpid/dispatch/amqp.h

```diff
@@ -19,6 +19,8 @@
  * under the License.
```

Comment:
https://app.travis-ci.com/github/apache/qpid-dispatch/jobs/559393998

```
18: ==13747==ERROR: LeakSanitizer: detected memory leaks
18: 
18: Direct leak of 13 byte(s) in 1 object(s) allocated from:
18:     #0 0x7f0b79f4730f in strdup (/usr/lib/x86_64-linux-gnu/libasan.so.2+0x6230f)
18:     #1 0x6260c6 in qd_strdup /home/travis/build/apache/qpid-dispatch/include/qpid/dispatch/ctools.h:266
18:     #2 0x62b336 in AMQP_rx_handler /home/travis/build/apache/qpid-dispatch/src/router_node.c:829
18:     #3 0x4bbb76 in do_receive /home/travis/build/apache/qpid-dispatch/src/container.c:219
18:     #4 0x4c05fd in qd_container_handle_event /home/travis/build/apache/qpid-dispatch/src/container.c:734
18:     #5 0x6444cf in handle /home/travis/build/apache/qpid-dispatch/src/server.c:1116
18:     #6 0x644697 in thread_run /home/travis/build/apache/qpid-dispatch/src/server.c:1141
18:     #7 0x524da1 in _thread_init /home/travis/build/apache/qpid-dispatch/src/posix/threading.c:172
18:     #8 0x7f0b798586b9 in start_thread (/lib/x86_64-linux-gnu/libpthread.so.0+0x76b9)
18: 
18: -----------------------------------------------------
18: Suppressions used:
18:   count      bytes template
18:       1         56 ^IoAdapter_init$
18:       2        128 sys_mutex
18:     561     859612 /libpython3.*.so
18: -----------------------------------------------------
18: 
18: SUMMARY: AddressSanitizer: 13 byte(s) leaked in 1 allocation(s).
18: <<<<
18: 
18: ----------------------------------------------------------------------
18: Ran 15 tests in 22.665s
18: 
18: FAILED (errors=1)
```

Some trace also https://app.travis-ci.com/github/apache/qpid-dispatch/jobs/559394005#L4544

Reply:
This does not seem to be Jira'd yet, assuming it is not covered by https://issues.apache.org/jira/browse/DISPATCH-830.

Reply:
@jiridanek ouch.  No, this triggered because there's a message leak somewhere in the system (see DISPATCH-1699).  This patch adds a new string field to the message structure.  When the message is freed this string is freed, so now when we leak a message this string leaks with it.
While we suppress the message leak itself, this new string is separate and needs it's own suppression.
I'll see if I can narrow the suppression to the qd_message_set_to_override_annotation call....